osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -133,7 +133,7 @@ public void SelectBeatmap(BeatmapInfo beatmap, bool animated = true)
 
         public void SelectNext(int direction = 1, bool skipDifficulties = true)
         {
-            if (groups.Count == 0)
+            if (groups.Count == 0 || groups.All(g => g.State == BeatmapGroupState.Hidden))

```

Comment:
Can do `!groups.Any(g => g.State != BeatmapGroupState.Hidden)`. Then you save yourself the `Count == 0` check.